apps/antalmanac/src/components/RightPane/AddedCourses/AddedCoursePane.tsx

```diff
@@ -73,12 +98,64 @@ function handleClear() {
     }
 }
 
+function ClearScheduleButton() {
+    return (
+        <Tooltip title="Clear Schedule">
+            <IconButton sx={buttonSx} onClick={handleClear}>
+                <DeleteOutline />
+            </IconButton>
+        </Tooltip>
+    );
+}
+
 function createCopyHandler(index: number) {
     return () => {
         copySchedule(index);
     };
 }
 
+function CopyScheduleButton() {
+    const [scheduleNames, setScheduleNames] = useState(AppStore.getScheduleNames());
+
+    useEffect(() => {
+        const handleScheduleNamesChange = () => {
+            setScheduleNames([...AppStore.getScheduleNames()]);
```

Comment:
**maybe** we could leave a note here explaining why the array has to be shallow copied? I didn't include one when I wrote it, but the reason is because React's state setter won't re-render the component if the value is the same, and the store internally pushes to the same array, so we need to clone it here.

```suggestion
            /**
             * A shallow copy needs to be made so the array reference is different and the component re-renders.
             */
            setScheduleNames([...AppStore.getScheduleNames()]);
```